cl8/users/forms.py

```diff
@@ -1,137 +1,186 @@
-from dal import autocomplete
-from django import forms
-from django.contrib.auth import forms as auth_forms
-from django.contrib.auth import get_user_model
-from django.contrib.auth.models import Group
-from django.core.exceptions import ValidationError
-from django.forms import ModelMultipleChoiceField
-from django.utils.translation import gettext_lazy as _
-from taggit import models as taggit_models
-
-from .importers import safe_username
-from .models import Profile
-
-User = get_user_model()
-
-
-class UserChangeForm(auth_forms.UserChangeForm):
-    class Meta(auth_forms.UserChangeForm.Meta):
-        model = User
-
-
-class UserCreationForm(auth_forms.UserCreationForm):
-    error_message = auth_forms.UserCreationForm.error_messages.update(
-        {"duplicate_username": _("This username has already been taken.")}
-    )
-
-    class Meta(auth_forms.UserCreationForm.Meta):
-        model = User
-
-    def clean_username(self):
-        username = self.cleaned_data["username"]
-
-        try:
-            User.objects.get(username=username)
-        except User.DoesNotExist:
-            return username
-
-        raise ValidationError(self.error_messages["duplicate_username"])
-
-
-class ProfileUpdateForm(forms.ModelForm):
-    name = forms.CharField()
-    # email = forms.EmailField()
-
-    tags = ModelMultipleChoiceField(
-        queryset=taggit_models.Tag.objects.all(),
-        widget=autocomplete.ModelSelect2Multiple(url="tag-autocomplete-with-create"),
-        required=False,
-    )
-
-    def save(self, commit=True):
-        """ """
-        self.instance.user.name = self.cleaned_data.get("name")
-        self.instance.user.save()
-
-        return super().save(commit=commit)
-
-    class Meta:
-        model = Profile
-        fields = [
-            "photo",
-            "name",
-            "phone",
-            "location",
-            "organisation",
-            "bio",
-            "tags",
-            "twitter",
```

Comment:
If this is used on an existing constellation, you'll like need a bit of code to handle this usage of different columns in the database

Reply:
Yes, we've changed to be 'generic social links', esp. as we no longer use or endorse the site formerly known as twitter. It autonames the link based on the root domain (e.g. linkedin, mastodon, etc)